### PR TITLE
[BWA-113] [BWA-114] [BWA-71 ] Issues With Alphabetical Ordering of TOTP Items

### PR DESCRIPTION
**Pull request by brant-livefront**
## 🎟️ Tracking

- [BWA-113](https://bitwarden.atlassian.net/browse/BWA-113) Editing items can cause them to jump out of their location in the list
- [BWA-114](https://bitwarden.atlassian.net/browse/BWA-114) Items not sorted alphabetically (e.g. "B" comes before "a")
- [BWA-71](https://bitwarden.atlassian.net/browse/BWA-71) When TOTP codes refresh, certain items "jump"

## 📔 Objective

In PR #190 we added an alphabetical ordering to the Item List. This fixed a number of cases where items would "jump" due to their being no order enforced. However, the ordering added was just simple comparison (e.g. `$0.name > $1.name`). The problem with this is that it sorts capital letters differently than lowercase - so "B" comes before "a". This results in the 2 issues above: the sorting is not the expected alphabetical, and editing an item named "A" to be named "a" means that it'd jump from the top of the list to below any capitalized items.

Elsewhere (in `AuthenticatorItemRepository`) we were using the more appropriate `localizedStandardCompare`. This was causing the third issue above - items sorted correctly at first launch could jump to the wrong location when we re-sort with the simple compare after refreshing their TOTP codes.

All three of these should now be fixed by moving to the `localizedStandardCompare` in both locations.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[BWA-113]: https://bitwarden.atlassian.net/browse/BWA-113?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[BWA-114]: https://bitwarden.atlassian.net/browse/BWA-114?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[BWA-71]: https://bitwarden.atlassian.net/browse/BWA-71?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: AuthenticatorShared/UI/Vault/ItemList/ItemList/ItemListProcessorTests.swift [hunk 1]
```diff
@@ -501,9 +501,14 @@ class ItemListProcessorTests: AuthenticatorTestCase { // swiftlint:disable:this
             ItemListItem.fixture(name: "Beta"),
             ItemListItem.fixture(name: "Delta"),
             ItemListItem.fixture(name: "Alpha"),
+            ItemListItem.fixture(name: "beta"),
         ]
         let resultSection = ItemListSection(id: "", items: results, name: "")
-        let resultSorted = ItemListSection(id: "", items: results.sorted(by: { $0.name < $1.name }), name: "")
+        let resultSorted = ItemListSection(
+            id: "",
+            items: results.sorted { $0.name.localizedStandardCompare($1.name) == .orderedAscending },
```
Comment (KatherineInCode) [inline AuthenticatorShared/UI/Vault/ItemList/ItemList/ItemListProcessorTests.swift:509]: 🤔 I know previously we did it algorithmically, but I think there could be value in explicitly stating what we expect the  order of the items to be. Not enough to not approve this, but just something I've been thinking about.

**Review by KatherineInCode — APPROVED** [review]

File: AuthenticatorShared/UI/Vault/ItemList/ItemList/ItemListProcessorTests.swift (hunk 1 above)
Comment (brant-livefront) [reply AuthenticatorShared/UI/Vault/ItemList/ItemList/ItemListProcessorTests.swift:509]: I'm cool with that. I had the exact same thought while I was doing this - i.e. if it had been manually assigned, maybe the test would have caught it originally. I kinda leaned towards keeping it algorithmic in the moment, but hearing you say that too pushes me back to manually specifying it. I'll throw that in a commit. 👍

Comment (KatherineInCode) [reply AuthenticatorShared/UI/Vault/ItemList/ItemList/ItemListProcessorTests.swift:509]: Definitely plays into the whole "testing behavior" vs "testing implementation" thing 🤔

File: AuthenticatorShared/UI/Vault/ItemList/ItemList/ItemListProcessorTests.swift [hunk 2]
```diff
@@ -501,9 +501,17 @@ class ItemListProcessorTests: AuthenticatorTestCase { // swiftlint:disable:this
...
             ItemListItem.fixture(name: "Delta"),
             ItemListItem.fixture(name: "Alpha"),
+            ItemListItem.fixture(name: "beta"),
+        ]
+        let resultsSorted = [
+            results[3], // Alpha
+            results[4], // beta
+            results[1], // Beta
+            results[2], // Delta
+            results[0], // Gamma
```
Comment (KatherineInCode) [inline AuthenticatorShared/UI/Vault/ItemList/ItemList/ItemListProcessorTests.swift:511]: 👍🏻 I appreciate the comments for context!

File: AuthenticatorShared/UI/Vault/ItemList/ItemList/ItemListProcessorTests.swift (hunk 1 above)
Comment (brant-livefront) [reply AuthenticatorShared/UI/Vault/ItemList/ItemList/ItemListProcessorTests.swift]: Updated in latest. 👍

**Review by KatherineInCode — APPROVED** [review]

**Pull request merged**